Hi @Jack-0-0 , thanks a lot for this! I merged the updated `dev` branch with the adjusted getters (that restore the column names of the new data snapshot) into this branch - hope that's OK! I then tried running `create_dataset.py` with both the updated data, and the previous version of the data. 

With the previous version it worked fine (I checked the test mode).

With the new data, it sill threw an error even after updating the getters. It looks like in the new version, there are unfortunately missing columns in the `crunchbase_people` table: specifically `location_id` and `country` columns.

```
/Users/karliskanders/Documents/innovation_sweet_spots/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/utils.py:92: FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().
  group_dummies = pd.get_dummies(cb_orgs["groups"].explode()).sum(level=0)
Traceback (most recent call last):
  File "innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py", line 325, in <module>
    typer.run(create_dataset)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/typer/main.py", line 859, in run
    app()
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py", line 172, in create_dataset
    )[PEOPLE_COLS]
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/pandas/core/frame.py", line 3464, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1)[1]
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/pandas/core/indexing.py", line 1314, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis)
  File "/Users/karliskanders/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/pandas/core/indexing.py", line 1377, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: "['person_country', 'location_id'] not in index"
```

That's annoying - I guess there might be two options. Either Crunchbase removed those columns for whatever reason, or perhaps they've been lost when Jack Vines updated the data? Perhaps you can reach out to Jack to check in about this?

Alternatively, perhaps you don't need the country and location_id data for people?

Finally, to answer your question about the lack of founder data for all companies: Perhaps we don't want to restrict this type of feature only to founders but include all staff listed on Crunchbase? Presumably, Crunchbase only lists senior staff members, so that could be quite relevant. I guess, it might be as easy as removing the criteria that the person needs to be a founder?

--- a/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
+++ b/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
@@ -75,6 +75,7 @@ DROP_COLS = [
     "first_funding_date_in_window",
     "last_funding_round_in_window",
     "latest_funding_date_in_window",
+    "last_funding_id_in_window",
     "org_id_x",
     "org_id_y",
     "org_id",

--- a/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
+++ b/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
@@ -80,16 +80,6 @@ DROP_COLS = [
     "org_id",
 ]
 
-PEOPLE_COLS = [
-    "person_id",
-    "person_country",
-    "location_id",
-    "org_id",
-    "is_founder",
-    "is_male_founder",
-    "is_female_founder",
-]
-
 
 def create_dataset(
     window_start_date: str = "01/01/2010",
@@ -164,15 +154,13 @@ def create_dataset(
         cb_people.pipe(utils.add_clean_job_title)
         .pipe(utils.add_is_founder)
         .pipe(utils.add_is_gender, gender="male")
-        .pipe(utils.add_is_gender, gender="female")
         .dropna(subset=["featured_job_organization_id"])
         .rename(
             columns={
                 "id": "person_id",
                 "featured_job_organization_id": "org_id",
-                "country": "person_country",
             }
-        )[PEOPLE_COLS]
+        )
         .reset_index(drop=True)
     )
 

--- a/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
+++ b/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py
@@ -9,6 +9,7 @@ On an M1 macbook it takes ~7 mins to run on the full dataset and ~1 min 30 secs 
 """
 import typer
 from innovation_sweet_spots import PROJECT_DIR
+from innovation_sweet_spots.getters import crunchbase
 from innovation_sweet_spots.getters.crunchbase import (
     get_crunchbase_funding_rounds,
     get_crunchbase_ipos,
@@ -24,6 +25,8 @@ from innovation_sweet_spots.pipeline.pilot.investment_predictions.create_dataset
 import pandas as pd
 from innovation_sweet_spots.analysis.wrangling_utils import CrunchbaseWrangler
 
+# Adjust the Crunchbase data snapshot path (ideally, should adapt the code to accommodate the newest snapshot)
+crunchbase.CB_PATH = crunchbase.CB_PATH.parents[0] / "cb_2021"
 
 KEEP_COLS = [
     "id",

--- a/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/utils.py
+++ b/innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/utils.py
@@ -716,8 +716,7 @@ def add_founders(cb_data: pd.DataFrame, org_id_founders: pd.DataFrame) -> pd.Dat
     ).fillna(
         {
             "founder_count": -1,
-            "female_founder_count": -1,
-            "male_founder_count": -1,
+            "male_founder_percentage": -1,
             "founder_max_degrees": -1,
             "founder_mean_degrees": -1,
         }

